src/project/internal/projectactionscontroller.cpp

```diff
@@ -938,15 +918,130 @@ void ProjectActionsController::warnPublishIsNotAvailable()
+    std::string body = trc("project",
+                           "This score has become corrupted and contains errors. "
+                           "To preserve your score, revert to the last saved version, or fix the errors manually. "
+                           "You can also save a copy and get help for this issue on musescore.org.");
+
+    IInteractive::ButtonDatas buttons;
+    buttons.push_back(interactive()->buttonData(IInteractive::Button::Cancel));
+
+    IInteractive::ButtonData saveCopyBtn(IInteractive::Button::CustomButton, trc("project", "Save a copy"), newlyCreated /*accent*/);
+    buttons.push_back(saveCopyBtn);
+
+    int defaultBtn = saveCopyBtn.btn;
+
+    IInteractive::ButtonData revertToLastSavedBtn(saveCopyBtn.btn + 1, trc("project", "Revert to last saved"),
+                                                  true /*accent*/);
+
+    int btn = interactive()->error(title, body, errorText, buttons, defaultBtn).button();
+
+    if (btn == saveCopyBtn.btn) {
+        saveProjectCopy(true);
+    } else if (btn == revertToLastSavedBtn.btn) {
+        revertCorruptedScoreToLastSaved();
+    }
+}
+
+void ProjectActionsController::warnScoreCorruptionLocallySave(const SaveLocation& location, const std::string& errorText, bool newlyCreated)
+{
+    std::string title = trc("project", "This score has become corrupted and contains errors");
+    std::string body = trc("project",
+                           "You can continue saving it locally, although the file may become unusable. "
+                           "To preserve your score, revert to the last saved version, or fix the errors manually. "
+                           "You can also get help for this issue on musescore.org.");
+
+    IInteractive::ButtonDatas buttons;
+    buttons.push_back(interactive()->buttonData(IInteractive::Button::Cancel));
+
+    IInteractive::ButtonData saveAnywayBtn(IInteractive::Button::CustomButton, trc("project", "Save anyway"), newlyCreated /*accent*/);
+    buttons.push_back(saveAnywayBtn);
+
+    int defaultBtn = saveAnywayBtn.btn;
+
+    IInteractive::ButtonData revertToLastSavedBtn(saveAnywayBtn.btn + 1, trc("project", "Revert to last saved"),
+                                                  true /*accent*/);
+    if (!newlyCreated) {
+        buttons.push_back(revertToLastSavedBtn);
+        defaultBtn = revertToLastSavedBtn.btn;
     }
 
-    interactive()->warning(trc("project/save", "Your score could not be saved"), msg);
+    int btn = interactive()->error(title, body, errorText, buttons, defaultBtn).button();
+
+    if (btn == saveAnywayBtn.btn) {
+        saveProjectAt(location, SaveMode::Save, true);
+    } else if (btn == revertToLastSavedBtn.btn) {
+        revertCorruptedScoreToLastSaved();
+    }
+}
+
+void ProjectActionsController::warnScoreCorruptionUponOpenning(const std::string& errorText)
+{
+    std::string title = trc("project", "Your score cannot be saved");
```

Comment:
warnScore**CorruptionUponOpenning** but "Your score cannot be **saved**"

Wrong text?